README.md

```diff
@@ -52,33 +114,47 @@ task :ci => ['engine_cart:generate'] do
 end
 ```
 
-And in your e.g. spec\_helper.rb (or rails\_helper.rb), initialize EngineCart:
+And in your engine's test framework configuration (e.g. spec\_helper.rb or rails\_helper.rb), initialize EngineCart:
 
 ```ruby
   require 'engine_cart'
   EngineCart.load_application!
 ```
 
-When you want to clean out and regenerate the test application, you can call:
+Your test files (e.g. spec files for your engine) can now be written as tests for a Rails application.
+
+## Gotcha
+
+If you have generated a test application, there is a Gemfile and Gemfile.lock associated with the testing app at spec/internal (or wherever you designated as the test app location).  If you then update *your* engine's Gemfile or gemspec, Bundler can get confused if it has conflicting information between your engine and the testing app.
+
+To fix this:
+
+1. clean out your testing app `$ rm spec/internal`  # rake may not be avail.
+2. remove your engine's Gemfile.lock `$ rm Gemfile.lock`  # not always necess.
+3. allow Bundler to resolve gem dependencies again `$ bundle install`
 
-```
-$ rake engine_cart:regenerate
-```
 
 ## Configuration
 
+### location of Rails testing app
+
 You can configure where the test app is created by setting the `ENGINE_CART_DESTINATION` env variable, e.g.:
 
-```ruby
-ENGINE_CART_DESTINATION="/tmp/generate-the-test-app-into-tmp-instead-of-your-app" rake ci
 ```
+ENGINE_CART_DESTINATION="/tmp/my_engines_test_app_here" rake engine_cart:generate
+```
+
+### adjusting generators for Rails testing app
+
+After creating the test application, EngineCart will run the test app generator (located in ./spec/test_app_templates/lib/generators). By default, it will attempt to run the `install` generator for your engine. If you do not have an `install` generator, or want to add additional steps (e.g. to install additional gems), you can add them to the `TestAppGenerator`.
 
-After creating the test application, Engine Cart will run the test app generator (located in ./spec/test_app_templates/lib/generators). By default, it will attempt to run the `install` generator for your engine. If you do not have an `install` generator, or want to add additional steps (e.g. to install additional gems), you can add them to the `TestAppGenerator`.
 
 ## Contributing
 
 1. Fork it
 2. Create your feature branch (`git checkout -b my-new-feature`)
 3. Commit your changes (`git commit -am 'Add some feature'`)
-4. Push to the branch (`git push origin my-new-feature`)
-5. Create new Pull Request
+4. Make sure to add tests for your changes.
+5. Make sure all engine_cart tests pass.
+6. Push to the branch (`git push origin my-new-feature`)
+7. Create a [Pull Request](https://help.github.com/articles/using-pull-requests/)
```

Comment:
Thanks. `engine_cart` is incredibly confusing (mostly because of the hoops bundler and rails make us jump through), and testing is non-obvious.